src/mds/Migrator.cc

```diff
@@ -1792,20 +1792,26 @@ void Migrator::encode_export_dir(bufferlist& exportbl,
     
     if (dn->get_linkage()->is_remote()) {
       // remote link
-      exportbl.append("L", 1);  // remote link
+      exportbl.append("l", 1);  // remote link
       
       inodeno_t ino = dn->get_linkage()->get_remote_ino();
       unsigned char d_type = dn->get_linkage()->get_remote_d_type();
+      ENCODE_START(2, 1, exportbl);
       encode(ino, exportbl);
       encode(d_type, exportbl);
+      encode(dn->get_alternate_name(), exportbl);
+      ENCODE_FINISH(exportbl);
       continue;
     }
 
     // primary link
     // -- inode
-    exportbl.append("I", 1);    // inode dentry
-    
+    exportbl.append("i", 1);    // inode dentry
+
+    ENCODE_START(2, 1, exportbl);
+    encode(dn->get_alternate_name(), exportbl);
```

Comment:
This means to make all the Dentry's realted fields together and then inode base. Since there has separate versions for Dentry and inode base, it should be okay. 

But to make it more comfortable I will fix it.